src/ingest/generation.rs

```diff
@@ -0,0 +1,42 @@
+use std::sync::atomic::{AtomicU32, AtomicUsize, Ordering};
+use std::sync::Arc;
+
+use chrono::{Datelike, Utc};
+
+pub struct SequenceGenerator {
+    counter: AtomicUsize,
+    last_reset_date: AtomicU32,
+}
+
+impl SequenceGenerator {
+    pub(crate) fn new() -> Self {
+        Self {
+            counter: AtomicUsize::new(0),
+            last_reset_date: AtomicU32::new(SequenceGenerator::get_current_date_time()),
+        }
+    }
+
+    pub(crate) fn generate_sequence_number(&self) -> usize {
+        let current_date_time = SequenceGenerator::get_current_date_time();
+        let last_reset_day = self.last_reset_date.load(Ordering::Acquire);
+
+        if last_reset_day == current_date_time {
+            return self.counter.fetch_add(1, Ordering::Relaxed);
+        }
+
+        self.counter.store(1, Ordering::Release);
+        self.last_reset_date
+            .store(current_date_time, Ordering::Release);
+        self.counter.fetch_add(1, Ordering::Relaxed)
```

Comment:
```
        self.counter.store(2, Ordering::Release);
        2
```

By using self.counter.store(2, Ordering::Release), you avoid the need to perform `store(1)` followed by `fetch_add(1)`, which essentially increments it to 2 anyway.

Besides, I think it's less confusing to use `1` instead of `2`.